src/dvclive/xgb.py

```diff
@@ -8,18 +9,35 @@
 
 class DVCLiveCallback(TrainingCallback):
     def __init__(
-        self, metric_data, model_file=None, live: Optional[Live] = None, **kwargs
+        self,
+        metric_data: Optional[str] = None,
+        model_file=None,
+        live: Optional[Live] = None,
+        **kwargs,
     ):
         super().__init__()
+        if metric_data is not None:
+            warn(
+                "`metric_data` is deprecated and will be removed",
+                category=DeprecationWarning,
+                stacklevel=2,
+            )
         self._metric_data = metric_data
         self.model_file = model_file
         self.live = live if live is not None else Live(**kwargs)
 
     def after_iteration(self, model, epoch, evals_log):
-        for key, values in evals_log[self._metric_data].items():
-            if values:
-                latest_metric = values[-1]
-            self.live.log_metric(key, latest_metric)
+        for name, subdir in (
+            [(self._metric_data, "")]
+            if self._metric_data
+            else zip(evals_log, evals_log)
+        ):
+            for key, values in evals_log[name].items():
+                if values:
+                    latest_metric = values[-1]
+                self.live.log_metric(
+                    f"{subdir}/{key}" if subdir else key, latest_metric
+                )
```

Comment:
Just minor aesthetics but I found this more clear than the `zip`:

```suggestion
        if self._metric_data:
            evals_log = {"": evals_log[self._metric_data]}
        for subdir, data in evals_log.items():
            for key, values in data.items():
                self.live.log_metric(
                    f"{subdir}/{key}" if subdir else key, values[-1]
                )
```

Unrelated, the `latest_metric` check was pointless

Reply:
> Unrelated, the latest_metric check was pointless

Yes, I was also wondering about that but didn't want to mix it with this PR. :laughing: